### ISSUE TITLE
Customized webhook Notification: clarify content of msg, heartbeatJSON and monitorJSON

### ISSUE THREAD

**Issue by clement-igonet** [louislam/uptime-kuma#4205]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification

### 🔖 Feature description

I'm interested in customizing webhook notification with suggested `msg, heartbeatJSON and monitorJSON`

However, their content is not clear to me. When I look at the source code, I can see part of indications, but I'm not sure I reach the whole content. More documentation would be helpful and welcome.

For instance, is there a way to get a unique ID / fingerprint for a given up and down sequence ?

- Indeed, it would help a lot when sending the notification up or down to third party services that need it to know how to close an incident by a unique up/down reference.``

### ✔️ Solution

Add inline documentation in webhook notification setup, or at least fill a content as a Test notification to see a whole `heartbeatJSON` or `monitorJSON` structure and content.

**Comment by CommanderStorm** [comment]
#3636 improved the documentation a bit, but currently this is difficult because some monitors have different information available.

Adding `monitorJSON` to the the test notification is something that we have not [got around to yet](https://github.com/louislam/uptime-kuma/blob/530c8e5328dd0cd3f75f3bee006887e4b4a775fd/CONTRIBUTING.md).

If you have more concrete ideas how we can do this we would be very open to this ^^

Note that webhooks can be tested using https://webhook.site/

**Comment by joy4eg** [comment]
>currently this is difficult because some monitors have different information available.

Yeah, that's understandable. But _I think_ they all (should?) share some common information/fields. Like `name`, `status` or `host`.
It would be really nice to see them (common fields) in the test notification run.

**Comment by unlimitedsola** [comment]
Stumbled upon this. I'm shocked by how little documentation on the detailed structure of the `heardbeatJSON` and `monitorJSON` is.

> Note that webhooks can be tested using https://webhook.site/

This would definitely be very helpful if the "test" button doesn't simply send `null` values.

**Comment by mansouryaacoubi** [comment]
There are many open issues discussing this, but still no definitive list of what `heartbeatJSON` and `monitorJSON` actually contain. So here is the short version:

Both JSON objects simply reflect the underlying **Monitor** and **Heartbeat** structures from Uptime Kuma’s _Common Data Structures_.
They are not special formats - just serialized versions of those objects.

**📌 Example: Monitor Object (partial)**
```json
{
  "id": 1,
  "name": "My Website",
  "type": "http",
  "url": "https://example.com",
  "method": "GET",
  "interval": 60,
  "retryInterval": 60,
  "resendInterval": 0,
  "maxretries": 0,
  "hostname": null,
  "port": null,
  "active": true,
  "tags": [
    {
      "tag_id": 1,
      "monitor_id": 1,
      "value": null,
      "name": "production",
      "color": "#059669"
    }
  ],
  "notificationIDList": { "1": true },
  "accepted_statuscodes_json": "[\"200-299\"]",
  "conditions": "[]"
  // ...plus other type-specific fields
}
```

**📌 Example: Heartbeat Object**
```json
{
  "monitorID": 1,
  "status": 1,          // 0=DOWN, 1=UP, 2=PENDING, 3=MAINTENANCE
  "time": "2023-10-27T10:30:00.123Z",
  "msg": "OK",
  "ping": 123,          // ms
  "important": true,    // indicates a status change
  "duration": 60,       // seconds since last heartbeat
  "localDateTime": "2023-10-27 12:30:00",
  "timezone": "Europe/Berlin",
  "retries": 0,
  "downCount": 0
}
```


The official structures referenced by these JSONs can be found here in the API Docs:
https://github.com/louislam/uptime-kuma/wiki/API-Documentation#common-data-structures

**Comment by CommanderStorm** [comment]
Fair, is now documented

https://github.com/louislam/uptime-kuma/wiki/Internal-API#common-data-structures

**Comment by laozhoubuluo** [comment]
> Fair, is now documented
> 
> https://github.com/louislam/uptime-kuma/wiki/Internal-API#common-data-structures

This document cannot be opened, it displays the following error message.

`You do not have permission to update this wiki.`

I tried searching within the Wiki but unfortunately couldn't find it. Could you please help me confirm if this document is still available? Thank you!

**Comment by CommanderStorm** [comment]
@louislam you seem to have deleted that page. This was not via a PR, but the editing interface, so I don't have any details why this happened..

If there is a reason, we can remove it again 😉

**Comment by louislam** [comment]
Yes, the doc is simply outdated, some info is not correct anymore. the doc was made in 2025 May, and no one maintain it since then.

I would recommend to directly ask any LLM to understand the internal API at this moment.
